src/unix_mode.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Stephanie Aelmore.
+// Conserve backup system.
+// Copyright 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022 Martin Pool.
+
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Discretionary Access Control permissions for archived files.
+//!
+//! On unix systems, the mode has 9 significant bits, divided into three classes,
+//! owner class, group class, and others class. The owner class permissions
+//! apply only to the owner, the group class permissions apply to members of the user's group,
+//! and the others class permissions apply to all other users.
+//!
+//! For each class, there are 3 bitflags: read, write, and execute. This is typically
+//! written as an octal number, such as 0o664, which means the user and group can
+//! both read and write, and other users can only read.
+//!
+//! The mode is also often presented as a string of characters, such as "rw-rw-r--",
+//! where each character represents one bit.
+//!
+//! On windows systems, files can be either read-only or writeable. For cross-compatibility,
+//! the mode is always stored using the unix format, where the read-only state is stored
+//! using the write bit in the user class.
+//! TODO: Properly implement and test Windows compatibility.
+//!
+use serde::{Deserialize, Serialize};
+use std::{fmt, fs::Permissions};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct UnixMode {
```

Comment:
I actually have an existing crate that implements dissecting and parsing similar to this. I'd rather add a dependency: https://github.com/sourcefrog/unix_mode, https://crates.io/crates/unix_mode

Reply:
Nice, I'll use that instead